diff --git a/llvm/utils/TableGen/GlobalISelEmitter.cpp b/llvm/utils/TableGen/GlobalISelEmitter.cpp
index 8d9ded1b2ac5..e8765e8bc3c9 100644
--- a/llvm/utils/TableGen/GlobalISelEmitter.cpp
+++ b/llvm/utils/TableGen/GlobalISelEmitter.cpp
@@ -770,9 +770,6 @@ Expected<InstructionMatcher &> GlobalISelEmitter::createAndImportSelDAGMatcher(
 
     if (Predicate.hasGISelPredicateCode()) {
       if (Predicate.usesOperands()) {
-        assert(WaitingForNamedOperands == 0 &&
-               "previous predicate didn't find all operands or "
-               "nested predicate that uses operands");
         TreePattern *TP = Predicate.getOrigPatFragRecord();
         WaitingForNamedOperands = TP->getNumArgs();
         for (unsigned i = 0; i < WaitingForNamedOperands; ++i)
